Add hack ifdef to assume assumption

lexs opened the pull request:
This is needed for now to handle old clients as they incorrectly didn't set the header but assumed resumption.

yschimke commented on src/Frame.cpp:
```diff
@@ -465,6 +465,9 @@ bool Frame_SETUP::deserializeFrom(std::unique_ptr<folly::IOBuf> in) {
     // TODO: Remove hack:
     // https://github.com/ReactiveSocket/reactivesocket-cpp/issues/243
+#ifdef HACK_ASSUME_RESUMPTION
+    header_.flags_ |= FrameFlags_RESUME_ENABLEö
```
WAT?!
`
FrameFlags_RESUME_ENABLEö`

lexs commented:
Updated, I switched the strategy to bump the version, this means we can assume resumption for older clients with version == 0 while everything else is running at version == 1.

lexs commented:
Reopened to trigger travis.

lehecka commented:
closing as this is not handled differently and the PR is obsolete.
